### ISSUE TITLE
Subtitles Button does not hide after remove subtitles tracks

### ISSUE THREAD

**Issue by mctep** [videojs/video.js#3890]
According to https://github.com/videojs/video.js/pull/3847

**Comment by mctep** [comment]
@gkatsev What to do with [ChaptersButton@update](https://github.com/videojs/video.js/blob/4fd902229fc1bd39f5abd8b033cfaf5a96c4de2b/src/js/control-bar/text-track-controls/chapters-button.js#L55)? As I understand there is should not be the logic with threshold in `ChaptersButton`. So we cannot simple move update method from `CaptionsButton` to `TextTrackButton` I think.

May be It better to make `createItems` for each button to return empty array in a specific cases? 
And I do not understand why `threshold` sets from 2 to 1 when [featuresNativeTextTracks]( https://github.com/videojs/video.js/blob/4fd902229fc1bd39f5abd8b033cfaf5a96c4de2b/src/js/control-bar/text-track-controls/captions-button.js#L58) enabled.

Other way I see it make yet another `TextTrackButtonWithThreshold` 😄  class.
Or set some `option` in each button constructors like `option.threshold` and use it in common `TextTrackButton.update` method. But it is not clear what to do with `featuresNativeTextTracks`.

**Comment by gkatsev** [comment]
Maybe a threshold of -1 disables this for the chapters button. I would prefer not to add another class :P

The reason why captions has a threshold of 1 for native and 2 for emulated is that we add an extra item to the menu for emulated captions for caption settings.

**Comment by mctep** [comment]
Good catch about `threshold = -1` 👍 

My third way with constructors is wrong because we have not info about tech in it and tech could be changed in player lifetime.

What do you think about this way:

```js
class SomeButton extends TextTrackButton {
  update() {
    this.threshold = someSpecificThresholdForThisButton();
    super.update();
  }
}
```

Or make `getMenuItemsTreshold` for `TextTrackButton` and redefine it in any `SomeButton`.

Or rewrite `createItems` 😄  that could return empty array without any additional buttons and any threshold logic when there are no tracks in track list. Why not?

**Comment by justinanastos** [comment]
This might be a totally naive suggestion, but it worked when modifying the source code locally. I did not think of edge cases, but this seems to work by checking to see if there one item left in the list and if it is an `instanceof OffTextTrackMenuItem`

#3993

**Comment by justinanastos** [comment]
I believe this issue has been resolved by #3993

**Comment by gkatsev** [comment]
#3993 and https://github.com/videojs/video.js/pull/4157, thanks @justinanastos!
